astroplan/scheduling.py

```diff
@@ -90,6 +90,80 @@ def from_exposures(cls, target, priority, time_per_exposure,
         return ob
 
 
+class Scorer(object):
+    """
+    Returns scores and score arrays from the evaluation of constraints on
+    observing blocks
+    """
+    def __init__(self, blocks, observer, schedule):
+        """
+
+        Parameters
+        ----------
+        blocks : list of `~astroplan.scheduling.ObservingBlock` objects
+            list of blocks that need to be scored
+        observer : `~astroplan.Observer`
+            the observer
+        schedule : `~astroplan.scheduling.Schedule`
+            The schedule inside which the blocks should fit
+        """
+        self.blocks = blocks
+        self.observer = observer
+        self.schedule = schedule
+        # the lists below should change if there are new constraints made
+        self.constraint_list = ['AltitudeConstraint', 'AirmassConstraint', 'AtNightConstraint',
```

Comment:
@kvyh - there may be a better way to do this. We can make the ```Constraint``` class have a MetaClass, which registers any subclass of ```Constraint``` that gets defined. This will then provide you with a list, or dictionary or whatever, of all defined ```Constraints```; be they ones that are implemented natively in astroplan or defined by the end user.

See http://www.effectivepython.com/2015/02/02/register-class-existence-with-metaclasses/ for details.